src/compiler/checker.ts

```diff
@@ -15195,6 +15201,23 @@ namespace ts {
             }
         }
 
+        function checkGlobalModuleExportDeclaration(node: GlobalModuleExportDeclaration) {
+            if (node.modifiers && node.modifiers.length) {
+                error(node, Diagnostics.Modifiers_cannot_appear_here);
+            }
+
+            if (node.parent.kind !== SyntaxKind.SourceFile) {
+                error(node, Diagnostics.Global_module_exports_may_only_appear_at_top_level);
+            }
+            else {
+                const parent = node.parent as SourceFile;
+                // Note: missing external module indicator case checked in binder

```

Comment:
I read this a few times before I realized it wasn't a strangely phrased TODO. Can you update this?